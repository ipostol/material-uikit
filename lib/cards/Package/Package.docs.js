'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _Package = require('./Package');

var _Package2 = _interopRequireDefault(_Package);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

exports.default = function () {
  return _react2.default.createElement(
    'div',
    { style: { height: '100%', display: 'flex', justifyContent: 'center', alignItems: 'center', backgroundColor: 'gray' } },
    _react2.default.createElement(_Package2.default, {
      cost: '\xA34000',
      description: 'Description',
      places: 'Places',
      resolutions: 'Resolutions',
      images: ['mhnzlkatwgr8nt0j88th', 'mhnzlkatwgr8nt0j88th', 'mhnzlkatwgr8nt0j88th'],
      header: 'Header',
      percent: '95%'
    })
  );
};