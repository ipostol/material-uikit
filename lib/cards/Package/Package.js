'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _Screen = require('../Screen');

var _Screen2 = _interopRequireDefault(_Screen);

var _Target = require('../../icons/Target');

var _Target2 = _interopRequireDefault(_Target);

var _Package = require('./Package.scss');

var _Package2 = _interopRequireDefault(_Package);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

var Package = function Package(_ref) {
  var header = _ref.header,
      percent = _ref.percent,
      props = _objectWithoutProperties(_ref, ['header', 'percent']);

  return _react2.default.createElement(
    _Screen2.default,
    props,
    _react2.default.createElement(
      'div',
      { className: _Package2.default.headerWrap },
      _react2.default.createElement(
        'div',
        { className: _Package2.default.header },
        header
      ),
      _react2.default.createElement(_Target2.default, { className: _Package2.default.icon }),
      _react2.default.createElement(
        'div',
        { className: _Package2.default.percent },
        percent
      )
    )
  );
};

Package.propTypes = {
  header: _propTypes2.default.string,
  percent: _propTypes2.default.string
};

exports.default = Package;