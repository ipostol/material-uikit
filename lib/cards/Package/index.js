'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = undefined;

var _Package = require('./Package');

var _Package2 = _interopRequireDefault(_Package);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

exports.default = _Package2.default;