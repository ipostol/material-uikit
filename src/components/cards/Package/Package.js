import React from 'react';
import PropTypes from 'prop-types';
import Screen from '../Screen';
import Target from '../../icons/Target';
import styles from './Package.scss';

const Package = ({ header, percent, ...props }) => (
  <Screen {...props}>
    <div className={styles.headerWrap}>
      <div className={styles.header}>{header}</div>
      <Target className={styles.icon} />
      <div className={styles.percent}>{percent}</div>
    </div>
  </Screen>
);

Package.propTypes = {
  header: PropTypes.string,
  percent: PropTypes.string,
};

export default Package;
