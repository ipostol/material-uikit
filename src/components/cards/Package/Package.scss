@import '~bidstack-theme/theme';

.headerWrap {
  display: flex;
  padding: 9px 16px;
  align-items: center;

  & + img {
    border-radius: 0;
  }
}

.header {
  font-size: $sizeBodyM;
  line-height: 1.47;
  margin-right: auto;
}

.icon {
  display: block;
  margin-right: 6px;
}

.percent {
  font-size: $sizeBodyM;
  line-height: 1.47;
  color: $fontColorLight2;
}
