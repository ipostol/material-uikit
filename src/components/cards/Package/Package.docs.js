import React from 'react';
import Package from './Package';

export default () => (
  <div style={{ height: '100%', display: 'flex', justifyContent: 'center', alignItems: 'center', backgroundColor: 'gray' }}>
    <Package
      cost="Â£4000"
      description="Description"
      places="Places"
      resolutions="Resolutions"
      images={['mhnzlkatwgr8nt0j88th', 'mhnzlkatwgr8nt0j88th', 'mhnzlkatwgr8nt0j88th']}
      header="Header"
      percent="95%"
    />
  </div>
);
