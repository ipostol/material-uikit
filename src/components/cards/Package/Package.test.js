import React from 'react';
import { shallow } from 'enzyme';
import Package from './Package';

describe('Package component', () => {

  it('should render correctly', () => {

    const props = {};

    const wrapper = shallow(
      <Package {...props} />,
    );

    expect(wrapper).toMatchSnapshot();

  });

});
