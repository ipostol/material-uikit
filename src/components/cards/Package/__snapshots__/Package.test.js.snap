// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Package component should render correctly 1`] = `
<Screen>
  <div
    className="headerWrap"
  >
    <div
      className="header"
    />
    <Target
      className="icon"
    />
    <div
      className="percent"
    />
  </div>
</Screen>
`;
